baseplate/observers/prometheus.py

```diff
@@ -121,15 +127,98 @@ def on_child_span_created(self, span: Span) -> None:
         else:
             observer = PrometheusClientSpanObserver()
 
-        observer.on_set_tag("protocol", self.protocol)
         span.register(observer)
 
 
 class PrometheusClientSpanObserver(SpanObserver):
     def __init__(self) -> None:
-        logger.debug("PrometheusClientSpanObserver not implemented")
+        self.tags: Dict[str, Any] = {}
+        self.start_time: Optional[int] = None
+        self.metrics: Optional[
+            Union[PrometheusHTTPClientMetrics]
+        ] = None  # Add PrometheusThriftClientMetrics when implemented
+
+    def on_set_tag(self, key: str, value: Any) -> None:
+        self.tags[key] = value
+
+    def get_prefix(self) -> str:
+        return f"{self.protocol}_client"
+
+    def get_labels(self) -> Dict[str, str]:
+        return self.tags
+
+    @property
+    def protocol(self) -> str:
+        return self.tags.get("protocol", "unknown")
+
+    def set_metrics_by_protocol(self) -> None:
+        if self.protocol == "http":
+            self.metrics = PrometheusHTTPClientMetrics()
+        elif self.protocol == "thrift":
+            # self.metrics = PrometheusThriftClientMetrics()
+            logger.debug("PrometheusThriftClientMetrics not implemented.")
+        else:
+            logger.warning(
+                "No valid protocol set for Prometheus metric collection, metrics won't be collected. Expected 'http' or 'thrift' protocol. Actual protocol: %s",
+                self.protocol,
+            )
```

Comment:
This log and the ones below probably don't warrant a 'warning' level, especially given that multiple are emitted on every span – perhaps 'debug' instead?

Reply:
Oh nvm I'm confused.
No these are bad. They shouldn't happen. We should always have a valid protocol as far as I know.